src/pal/src/misc/miscpalapi.cpp

```diff
@@ -454,3 +454,159 @@ EnsureOpenSslInitialized()
     return dwRet;
 }
 
+/*++
```

Comment:
@kangaroo I have bumped into this problem in the past and no attributes helped. When 
1) there is a .o file inside of a static library and 
2) this library is linked as a part of a dynamic library and
3) no variable or function in the .o file are referenced from anywhere
then the linker ommits the .o file completely.
It is enough to add e.g. a global variable to the source file that's compiled to the .o and reference it from other place and then the .o file is included.